pkg/cmd/pr/checkout/checkout.go

```diff
@@ -240,6 +240,24 @@ func localBranchExists(client *git.Client, b string) bool {
 	return err == nil
 }
 
+// Warn if the pull request and the remote branch have diverged.
+func warnIfDiverged(opts *CheckoutOptions, pr *api.PullRequest) {
+	if len(pr.Commits.Nodes) == 0 {
+		return
+	}
+
+	localBranchLastCommit, err := opts.GitClient.LastCommit(context.Background())
+	if err != nil {
+		return
+	}
```

Comment:
**Swallowing this error is a bit concerning**

I understand this function and the ones it is called from aren't designed to bubble errors up to the calling code, so it isn't pretty in this form.

Stepping back, I wonder if `warnIfDiverged` should be moved up the stack within `checkoutRun` instead; essentially we just need to call this whenever `checkout` is being called without the force flag, which should allow us to bubble up the error and minimize the places it is called 🤔 

https://github.com/cli/cli/blob/c15da4e512073913b3f418dcf461eb9ccf0acb77/pkg/cmd/pr/checkout/checkout.go#L109-L126